src/rebase.c

```diff
@@ -1327,6 +1327,22 @@ int git_rebase_finish(
 	return error;
 }
 
+const git_oid *git_rebase_orig_head_id(git_rebase *rebase) {
+	return (rebase->type == GIT_REBASE_TYPE_MERGE) ? &rebase->orig_head_id : NULL;
+}
+
+const char *git_rebase_orig_head_name(git_rebase *rebase) {
+	return (rebase->type == GIT_REBASE_TYPE_MERGE) ? rebase->orig_head_name : NULL;
+}
+
+const git_oid *git_rebase_onto_id(git_rebase *rebase) {
```

Comment:
That's minor, but I don't like how those two names parse: "onto" makes me expect that something would happen, but those are accessors…

What about `git_rebase_ref_(orig_head|onto)(_id)`. Alternatively, maybe the implementation can be changed to copy the annotated commits instead of the current oid/name split, which would make the final interface a 2-function `git_rebase_annotated_(orig_head|onto)` ?

Reply:
Personally I would let the caller decide if he wants to create a new object for the annotated commit. I'd go with `git_rebase_(orig_head|onto)_ref` and `git_rebase_(orig_head|onto)_id`.

Reply:
I for one am mainly interested in the name/ref alone.

Reply:
Oh I see... you mean storing `git_annotated_commit *branch` as `orig_head` and `git_annotated_commit *onto` as `onto` instead of separate values in `rebase_init_merge`?

Reply:
Ok, now returning `annotated_commit`s. Also has the benefit of returning fully qualified ref names.

Reply:
But i dont know... simplest would be to just omit annotated commits and oids and just return the ref names. If someone wants the ref, they could always dwim it themselves.